src/cmakeTaskProvider.ts

```diff
@@ -373,10 +373,17 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
             const result: proc.ExecutionResult = await proc.execute(cmakePath, args, this, this.options).result;
             if (result.retc) {
                 this.writeEmitter.fire(localize("build.finished.with.error", "{0} finished with error(s).", taskName) + endOfLine);
-            } else if (result.stderr && !result.stdout) {
-                this.writeEmitter.fire(localize("build.finished.with.warnings", "{0} finished with warning(s).", taskName) + endOfLine);
-            } else if (result.stdout && result.stdout.includes("warning")) {
+                if (doCloseEmitter) {
+                    this.closeEmitter.fire(result.retc);
+                }
+                vscode.commands.executeCommand("workbench.actions.view.problems")
```

Comment:
I don't think this is the correct approach.  To get the problems to focus, you need a problem matcher and then VS Code will do the right thing based on your settings in the task.  Add `"problemMatcher": "$msCompile"` to your task to make this work automatically.